test/testsuite/base.jl

```diff
@@ -283,13 +283,13 @@ end
             # due to different definition of `Int` type
             # print([1]) shows as [1] on 64bit but Int64[1] on 32bit
             msg = showstr(A)
-            @test msg == "[1]" || msg == "Int64[1]"
+            @test occursin("[1]", msg) || occursin("Int64[1]", msg)
```

Comment:
Multiple `occursin`s with strings that are substrings of each other isn't very useful.

Reply:
This is true. Maybe these should detect the brackets which this PR adds, at least this will fail before:
```suggestion
            @test occursin("([1])", msg) || occursin("(Int64[1])", msg)
```

Reply:
These tests fail on CI, and running the package's tests locally, on Julia <= 1.8. 

But what they aim to test seems to work fine in isolation, and with just this `@testset`. Any idea what's wrong?
```julia
julia> using JLArrays

julia> jl([1]), 2
(JLArray([1]), 2)

julia> # compact=false to avoid type aliases
       replstr(x, kv::Pair...) = sprint((io,x) -> show(IOContext(io, :compact => false, :limit => true, :displaysize => (24, 80), kv...), MIME("text/plain"), x), x)
replstr (generic function with 1 method)

julia> let AT = JLArray
               @testset "showing" begin
                   # vectors and non-vector arrays showing
                   # are handled differently in base/arrayshow.jl
                   A = AT(Int64[1])
                   B = AT(Int64[1 2;3 4])
                   msg = replstr(A)
                   @test occursin(Regex("^1-element $AT{Int64,\\s?1.*}:\n 1\$"), msg)
                   # # result of e.g. `print` differs on 32bit and 64bit machines
                    # due to different definition of `Int` type
                    # print([1]) shows as [1] on 64bit but Int64[1] on 32bit
                    msg = showstr(A)
                    @test occursin("([1])", msg) || occursin("(Int64[1])", msg)

                    msg = replstr(B)
                    @test occursin(Regex("^2×2 $AT{Int64,\\s?2.*}:\n 1  2\n 3  4\$"), msg)

                    msg = showstr(B)
                    @test occursin("([1 2; 3 4])", msg) || occursin("(Int64[1 2; 3 4])", msg)

                    # the printing of Adjoint depends on global state
                    msg = replstr(A')
                   @test occursin(Regex("^1×1 Adjoint{Int64,\\s?$AT{Int64,\\s?1.*}}:\n 1\$"), msg) ||
                       occursin(Regex("^1×1 LinearAlgebra.Adjoint{Int64,\\s?$AT{Int64,\\s?1.*}}:\n 1\$"), msg) ||
                       occursin(Regex("^1×1 adjoint\\(::$AT{Int64,\\s?1.*}\\) with eltype Int64:\n 1\$"), msg)
               end
           end
Test Summary: | Pass  Total
showing       |    5      5
Test.DefaultTestSet("showing", Any[], 5, false, false)

julia> VERSION
v"1.6.0"
```

Reply:
These tests fail on CI, and running the package's tests locally, on Julia <= 1.8. 

But what they aim to test seems to work fine in isolation, and with just this `@testset`. Any idea what's wrong?
```julia
julia> using JLArrays

julia> jl([1]), 2
(JLArray([1]), 2)

julia> # compact=false to avoid type aliases
       replstr(x, kv::Pair...) = sprint((io,x) -> show(IOContext(io, :compact => false, :limit => true, :displaysize => (24, 80), kv...), MIME("text/plain"), x), x)
replstr (generic function with 1 method)

julia> let AT = JLArray
               @testset "showing" begin
                   # vectors and non-vector arrays showing
                   # are handled differently in base/arrayshow.jl
                   A = AT(Int64[1])
                   B = AT(Int64[1 2;3 4])
                   msg = replstr(A)
                   @test occursin(Regex("^1-element $AT{Int64,\\s?1.*}:\n 1\$"), msg)
                   # # result of e.g. `print` differs on 32bit and 64bit machines
                    # due to different definition of `Int` type
                    # print([1]) shows as [1] on 64bit but Int64[1] on 32bit
                    msg = showstr(A)
                    @test occursin("([1])", msg) || occursin("(Int64[1])", msg)

                    msg = replstr(B)
                    @test occursin(Regex("^2×2 $AT{Int64,\\s?2.*}:\n 1  2\n 3  4\$"), msg)

                    msg = showstr(B)
                    @test occursin("([1 2; 3 4])", msg) || occursin("(Int64[1 2; 3 4])", msg)

                    # the printing of Adjoint depends on global state
                    msg = replstr(A')
                   @test occursin(Regex("^1×1 Adjoint{Int64,\\s?$AT{Int64,\\s?1.*}}:\n 1\$"), msg) ||
                       occursin(Regex("^1×1 LinearAlgebra.Adjoint{Int64,\\s?$AT{Int64,\\s?1.*}}:\n 1\$"), msg) ||
                       occursin(Regex("^1×1 adjoint\\(::$AT{Int64,\\s?1.*}\\) with eltype Int64:\n 1\$"), msg)
               end
           end
Test Summary: | Pass  Total
showing       |    5      5
Test.DefaultTestSet("showing", Any[], 5, false, false)

julia> VERSION
v"1.6.0"
```